### DISCUSSION TITLE
how to set tiles file path -backend

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#1791] [Q&A] [answered]
i open the bim file from Snapshot by imodejs,and load on the web front by webbrower,and the backend create bim.Tiles  file
how i can set the folder path to store the .Tiles file ?

**Comment by pmconne** [comment]
The .tiles file is currently always created in the same directory as the .bim file - it is not configurable. What problem is this causing you?

**Reply by paprophet** [reply]
because the backend process lock the .Tiles file usually  when i refresh the webbrower frequently.and then  webbrower can not show the model  only if  i must restart the backend .

**Comment by pmconne** [comment, marked as answer]
The .tiles tile cache is intended for use in single-client applications only (i.e., electron or mobile apps). A multi-user web app should use a [shared tile cache service](https://github.com/imodeljs/imodeljs/blob/c889d837b9f034d28da4e99b93d8a05891cdfe72/core/backend/src/IModelHost.ts#L328) instead.

**Reply by paprophet** [reply]
i set this,but the d:\temp3 dictory is empty after open the .bim file by web brower, and create .Tiles exist.


const config = new IModelHostConfiguration(); 
  
    config.crashReportingConfig={
      crashDir: "d:\\temp3",  
      enableCrashDumps:true,
      enableNodeReport:true,
    };  
    await IModelHost.startup(config);

**Reply by pmconne** [reply]
The crashReportingConfig has nothing to do with the tile cache. If you need help with crash reporting, please post a separate question. You cannot configure the directory in which the .tiles file is stored.

**Reply by paprophet** [reply]
ok thank you !

**Reply by paprophet** [reply]
how to create tileCacheService  by the bim file  (create by shotshop iTwin Snapshot） ??

**Reply by pmconne** [reply]
Two options:
1. Use an AliCloud or Azure storage account. Provide the account credentials via [IModelHostConfiguration.tileCacheCredentials](https://www.itwinjs.org/reference/imodeljs-backend/imodelhost/imodelhostconfiguration/tilecachecredentials/) when starting up the backend.
2. Write your own implementation of a CloudStorageService that writes the tiles to some location on disk, or wherever else you want

These are beta APIs and I can't give you a good example of #2 above. display-test-app contains a very basic, probably buggy [implementation](https://github.com/imodeljs/imodeljs/blob/master/test-apps/display-test-app/src/backend/FakeTileCacheService.ts) that writes the tiles to a location on disk. I offer this not as an example, but to show the basics of what you would need to implement.

@swbsi may be able to assist further, if you can be more specific about what you're trying to accomplish.
